src/rust/engine/fs/store/src/local.rs

```diff
@@ -282,12 +641,28 @@ impl ByteStore {
     items: Vec<(Fingerprint, Bytes)>,
     initial_lease: bool,
   ) -> Result<(), String> {
-    let dbs = match entry_type {
-      EntryType::Directory => self.inner.directory_dbs.clone(),
-      EntryType::File => self.inner.file_dbs.clone(),
-    };
+    let mut fsdb_items = vec![];
+    let mut lmdb_items = vec![];
+    for (fingerprint, bytes) in items {
+      if entry_type == EntryType::File && ByteStore::should_use_fsdb(bytes.len()) {
```

Comment:
Nit: the `should_use_fsdb` method should probably take the `EntryType` as well, since effectively every usage needs to do the `entry_type == EntryType::File` comparison as well.